### ISSUE TITLE
Deduplicate `getNextBest` and `dislodgeOffer` if possible

### ISSUE THREAD

**Issue by adhusson** [mangrovedao/mangrove-core#474]
Try to deduplicate `dislodgeOffer`, `getNextBest`, and the last part of `writeOffer` (all subtly different). Especially implement a `displaceOffer` function that takes into account the fact that some removed offers will be moved elsewhere, should increase efficiency. This is partially solved: `dislodgeOffer` has an additional parameter that tells it to forgo updating the branch cache in `local`. But they are still all different functions and having a master implem would probably be safer (although less gas-efficient)

**Comment by espendk** [comment]
The tests in PR #486 should be helpful in ensuring that this deduplication is sound.

**Comment by espendk** [comment]
Oh, and the bug identified in #486  should probably be fixed before starting work on this task.

**Comment by adhusson** [comment]
Deduplication attempted in #509 and aborted, resulting code was too unwieldy & `getNextBest` has marketOrder-specific optimizations that could not be fully retained when merged into `dislodgeOffer`.
